src/Scriban/Syntax/ScriptNode.cs

```diff
@@ -104,6 +104,14 @@ public static bool HasTriviaEndOfStatement(this ScriptNode node, bool before)
             }
             return false;
         }
+
+        public static TNode WithTriviaAndSpanFrom<TNode>(this TNode node, ScriptNode sourceNode)
+            where TNode : ScriptNode
+        {
+            node.Trivias = sourceNode?.Trivias;
+            node.Span = sourceNode?.Span ?? default;
+            return node;
+        }
```

Comment:
Initially I had made this method internal, but when writing the tests I realized it could be useful to people who implement custom visitors.